application/src/main/java/io/redlink/smarti/webservice/LegacyConversationWebservice.java

```diff
@@ -0,0 +1,277 @@
+/*
+ * Copyright 2017 Redlink GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package io.redlink.smarti.webservice;
+
+import io.redlink.smarti.api.QueryBuilder;
+import io.redlink.smarti.api.StoreService;
+import io.redlink.smarti.model.*;
+import io.redlink.smarti.model.config.ComponentConfiguration;
+import io.redlink.smarti.model.config.Configuration;
+import io.redlink.smarti.model.result.Result;
+import io.redlink.smarti.services.ClientService;
+import io.redlink.smarti.services.ConfigurationService;
+import io.redlink.smarti.services.ConversationService;
+import io.redlink.smarti.services.QueryBuilderService;
+import io.redlink.smarti.utils.ResponseEntities;
+import io.redlink.smarti.webservice.pojo.QueryUpdate;
+import io.redlink.smarti.webservice.pojo.TemplateResponse;
+import io.swagger.annotations.Api;
+import io.swagger.annotations.ApiOperation;
+import io.swagger.annotations.ApiResponse;
+import io.swagger.annotations.ApiResponses;
+import org.bson.types.ObjectId;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.util.MimeTypeUtils;
+import org.springframework.web.bind.annotation.*;
+
+import java.io.IOException;
+import java.util.Map.Entry;
+import java.util.Objects;
+import java.util.Optional;
+
+
+/**
```

Comment:
Nope, I just keep the old implementation as a reference during implementation. This class (together with the `RocketChatEndpoint`) will be removed.

Reply:
What's about Solr's strategy of deprication: keep that implementation, declare it as deprecated and then remove it with the next major release?

Reply:
The point is: with the changes introduced with the new API (esp. with the changes in the data-model), the `LegacyConversationWebservice` will not work any more.
We would need invest additional effort to keep the old web-services functional.

Waiting for the next major release is not necessary, since we're still in a [`0.x`](http://semver.org/spec/v2.0.0.html#spec-item-4) version.

Reply:
@ja-fra I'm fine with deleting that at the current level of "maturity" without prior depreciation 👍